Shop/main.py

```diff
@@ -0,0 +1,123 @@
+"""Валидация полученного json через схему и запись в базу."""
+
+import jsonschema
+import psycopg2
+import json
+from config import config
+
+
+def read_json(file: str) -> dict:
+    """Чтение из файла в словарь."""
+    with open(file, 'r', encoding='utf-8') as json_file:
+        data = json.load(json_file)
+        return data
+
+
+input_json = read_json("input_data.json")
+validation_schema = read_json("goods_schema.json")
+
+
+def validate(file: dict, schema: dict) -> bool:
+    """Валидация входного файля через схему."""
+    try:
+        jsonschema.validate(file, schema)
+        return True
+    except jsonschema.exceptions.ValidationError:
+        return False
+
+
+def prepare_tuple(input_dict: dict) -> tuple:
+
+    """Убирает вложенность и формирует кортежи для последующей записи в БД.
+    Для каждой таблицы создаётся отдельный кортеж.
+    """
+    goods = []
+    shops_goods = []
+    for key, value in input_dict.items():
+        if key == "id":
+            goods.append(value)
+            shops_goods.append(value)
+        if key == "name":
+            goods.append(value)
+        if isinstance(value, dict):
+            sub_element = value
+            for sub_key, sub_value in sub_element.items():
+                if sub_key in ["width", "height"]:
+                    goods.append(sub_value)
+        if isinstance(value, list):
+            for i in range(len(value)):
```

Comment:
if isinstance(value, list):
            for sub_element in value:

Reply:
fixed